data/lib/lib_copy_spider.lua

```diff
@@ -0,0 +1,65 @@
+function create_copy_spidertron(spidertron_name)
+    local constructron_name = "constructron-" .. spidertron_name
+    local constructron = table.deepcopy(data.raw["spider-vehicle"][spidertron_name])
+    constructron.name = constructron_name
+    constructron.collision_mask = {
+        "water-tile",
+        "colliding-with-tiles-only",
+        "not-colliding-with-itself"
+    }
+    constructron.minable.result = constructron_name
+
+    local constructron_item = {
+        name = constructron_name,
+        order = "b[personal-transport]-c[spidertron]-a[spider]b",
+        place_result = constructron_name,
+        stack_size = 1,
+        subgroup = "transport",
+        type = "item-with-entity-data"
+    }
+
+    if constructron.icons then
+        local item_icons = constructron.icons
+        table.insert(item_icons, 1, {
+            icon = "__Constructron-Continued__/graphics/icon_texture.png",
+            icon_size = 256,
+            scale = 0.25
+        })
+        constructron_item.icons = item_icons
```

Comment:
This seems weird. It will insert the `icon_texture` to `constructron.icons` as well, you would need to `table.deepcopy` it into `item_icons`.

That said, I do think you need to ensure here that the `icon_texture` (which I assume is that construction block around the spider) is added to both the item _and_ the entity icon.

Reply:
Since I'm not using `table.deepcopy` for duplicating `constructron.icons` `table.insert` will affect both the entity and item. I'll make that step more obvious by inserting straight into constructron.icons. 
This doesn't happen for the entity when the constructron has a single icon, which will be fixed.